src/python/pants/core/util_rules/subprocess_environment.py

```diff
@@ -67,11 +67,16 @@ class SubprocessEnvironmentVars:
 
 
 @rule
-def get_subprocess_environment(
-    subproc_env: SubprocessEnvironment, env: CompleteEnvironment
+async def get_subprocess_environment(
```

Comment:
See comment below - but it ripples here to ditto - this need not be async or even a rule, it could just be a function.